### PR TITLE
Add message size limit on event emission in local server

### LINKED ISSUE #8888: Local Server Max Message Size Enforcement
There are max message size limitations to using websockets, but local server and its driver do not use websockets. Since local server is frequently used in test scenarios it can lead to things working in test, but not in production. We should add an artificial limitation to the local server or driver that enforces the max message size

### PR DESCRIPTION
**Pull request by justus-camp**
# [Task 278](https://dev.azure.com/fluidframework/internal/_workitems/edit/278)

## Description
The local server implementation currently doesn't have a max message size implemented. There is a 1MB message size limit (more info in #7599), so this can lead to situations where tests pass on the local server implementation but not in production. 

I've taken a simple approach of just stringifying the payload, checking if the length is greater than 1MB, and disconnecting the socket if it is. 

Fixes #8888

### CODE REVIEW

File: server/routerlicious/packages/local-server/src/localWebSocketServer.ts [hunk 1]
```diff
@@ -37,6 +37,11 @@ export class LocalWebSocket implements core.IWebSocket {
     }
 
     public emit(event: string, ...args: any[]) {
+        // Disconnect from the "socket" if the message is greater than 1MB
+        if (JSON.stringify(args).length > 1e6) {
+            this.disconnect();
```
Comment (GaryWilber) [inline server/routerlicious/packages/local-server/src/localWebSocketServer.ts:42]: might want to add a `return;` after this so the emit line won't get called?

Comment (justus-camp) [reply server/routerlicious/packages/local-server/src/localWebSocketServer.ts:42]: good call, for some reason I had decided the return wasn't necessary. looks fairly obvious to me now that it is 😄

**Review by GaryWilber — APPROVED** [review]

File: server/routerlicious/packages/local-server/src/localWebSocketServer.ts (hunk 1 above)
Comment (vladsud) [reply server/routerlicious/packages/local-server/src/localWebSocketServer.ts:42]: @GaryWilber, @justus-camp - this change causes trouble. What I see is that local tests (running in-memory server) fail for large payloads as the "server" accumulates a ton of ops (each of them is smaller than 1Mb) and hits that check, getting into infinite loop of disconnects & reconnects (maybe it's not infinite, but test stops making any progress).
And if I comment this code out, everything works fine.

Repro is simple - a UT that sends a ton of largish ops (anything in 250K - 800K range). Batching is done in runtime (not using protocol capability), so every op shows up as one op in a batch.

@NicholasCouri - FYI

Comment (vladsud) [reply server/routerlicious/packages/local-server/src/localWebSocketServer.ts:42]: I have a feeling that what you wanted is on inbound direction (client side) to put that limit, not on outbound (service side). But I think same class is used for both for in-memory server, and it screws everything.

**Pull request merged**